### ISSUE TITLE
nix-installer uses broken nixpart-0.4.1, deployment fails

### ISSUE THREAD

**Issue by braunse** [NixOS/nixops-hetzner#3]
## What I want to do:

I tried getting started with nixops by deploying a simple machine definition to a server I already have.

My config:

```
let
  partitiondef = ''
    clearpart --all --initlabel --drives=sda,sdb

    part --size=1024 --ondisk=sda raid.1
    part --size=1024 --ondisk=sdb raid.2

    raid --device=boot-r1 --fstype=ext4 --level=1 --fsoptions=nosuid,noexec /boot raid.1 raid.2

    part --size=65536 --label=swap1 --fstype=swap --ondisk=sda swap
    part --size=65536 --label=swap2 --fstype=swap --ondisk=sdb swap

    part --size=65536 --ondisk=sda btrfs.1
    part --size=65536 --ondisk=sdb btrfs.2

    btrfs / --label root --data=1 --metadata=1 btrfs.1 btrfs.2

    part --grow --ondisk=sda btrfs.3
    part --grow --ondisk=sdb btrfs.4

    btrfs /vstor --label vstor --data=1 --metadata=1 btrfs.3 btrfs.4
  '';
  vmserverskel = 
    { deployment.targetEnv = "hetzner";
      deployment.hetzner.partitions = partitiondef;
      deployment.hetzner.createSubAccount = false;
    };
in
{
  dibs1 =
    { config, pkgs, ...}:
    vmserverskel //
    {
      services.nginx.enable = true;
      services.openssh.enable = true;

      deployment.targetEnv = "hetzner";
      deployment.hetzner.partitions = partitiondef;
      deployment.hetzner.createSubAccount = false;
      deployment.hetzner.mainIPv4 = "333.333.333.333";
      deployment.hetzner.robotUser = "<<USERACCOUNT>>";
    };
}
```

## What happens

Deployment fails immediately after the rescue system comes up, with the message:

```
building Nix bootstrap installer... error: Package ‘nixpart-0.4.1’ in /nix/var/nix/profiles/per-user/root/channels/nixos/pkgs/tools/filesystems/nixpart/0.4/default.nix:71 is marked as broken, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

(use '--show-trace' to show detailed location information)
```

## What I expected to happen

NixOS deployed to the server given.

## What I think might be the problem

nixpart-0.4.1 depends on a version of pykickstarter that will not build correctly in the current NixOS packages. nixpart-1.0.0 installs into my env without problems.

## Possible solution

hetzner-bootstrap-installer should depend on nixpart 1.0.0 instead of the broken version.

**Comment by aszlig** [comment]
> nixpart-0.4.1 depends on a version of pykickstarter that will not build correctly in the current NixOS packages. nixpart-1.0.0 installs into my env without problems.

Note that nixpart 1.0.0 is unreleased and the current package is a [stub](https://github.com/NixOS/nixpkgs/commit/16371fce91709932df51d276aaac4f90dc004df2#diff-beb432de953ab427ae94391538500729) and intentionally marked as broken as well.

So the question is now: @lheckemann Why is 0.4.1 marked as broken as well (see https://github.com/NixOS/nixpkgs/commit/5aa4b1994629ed0701b8b6d6e094da9d7d912721)?

**Comment by lheckemann** [comment]
That was probably one of several mistakes we made in that mark-broken spree, sorry!

**Comment by lheckemann** [comment]
Fixed: 75667c2916d (master), d76f279a65e (19.09)

**Comment by rbvermaa** [comment]
The build was actually broken, I pushed a fix for the nixpart0 build on 19.09 and master. Testing the hetzner bootstrap now.

**Comment by rbvermaa** [comment]
Everything is working now with the fixed on master/19.09. Closing.
